R/cdfPlot.R

```diff
@@ -24,117 +21,226 @@
-                    plotType = "canvasXpress",
-                    pvalCol = "P.Value",
-                    pThreshold = 0.01,
-                    xlab = NULL,
-                    ylab = NULL,
-                    title = NULL,
-                    insetTitle = NULL,
-                    symbolSize = c(2, 1),
-                    symbolShape = c(20, 20),
-                    symbolColor = c("red3", "deepskyblue4"),
-                    symbolFill = c("red3", "deepskyblue4"),
-                    alpha = 1,
-                    referenceLine = NULL,
+                    plotType       = "canvasXpress",
+                    pvalCol        = "P.Value",
+                    pThreshold     = 0.01,
+                    xlab,
+                    ylab,
+                    title          = NULL,
+                    insetTitle     = NULL,
+                    symbolSize     = c(4, 3),
+                    symbolShape    = c("circle", "circle"),
+                    symbolColor    = c("red3", "deepskyblue4"),
+                    transparency   = 0.7,
+                    referenceLine  = NULL,
                     refLineThickness = 3,
-                    legendPosition = "right",
-                    viewportX = 0.15,
-                    viewportY = 0.85,
-                    viewportWidth = 0.35,
-                    pvalMax = 0.10,
-                    printPlot = TRUE,
-                    footnote,
-                    footnoteSize = 3,
-                    footnoteColor = "black",
-                    footnoteJust = 1) {
-
-    assertthat::assert_that(pvalCol %in% colnames(contrastDF),
-                            msg = "Specified pvalCol not found in the supplied dataframe (contrastDF).")
-    assertthat::assert_that(plotType %in% c("canvasXpress", "ggplot"),
-                            msg = "Plot type must be either canvasXpress or ggplot.")
-
-    if (!missing(symbolSize) || !missing(symbolShape) || !missing(symbolColor) || !missing(symbolFill)) {
-        assertthat::assert_that(!length(symbolSize) == 2,
-                                !length(symbolShape) == 2,
-                                !length(symbolColor) == 2,
-                                !length(symbolFill) == 2,
-                                msg = "All specified symbol arguments must be of length 2, including symbolSize, symbolShape, symbolColor, and symbolFill.")
+                    viewportX      = 0.15,
+                    viewportY      = 0.85,
+                    viewportWidth  = 0.35,
+                    pvalMax        = 0.10,
+                    footnote) {
+
+    assertthat::assert_that(!missing(contrastDF),
+                            !is.null(contrastDF),
+                            "data.frame" %in% class(contrastDF),
+                            nrow(contrastDF) > 0,
+                            msg = "contrastDF must be specified as dataframe with a p-value column.")
+
+    assertthat::assert_that(!is.null(plotType),
```

Comment:
as I recall from last discussion with Connie, we agrees on setting plotType to default in case there is an issue and generates warning like other parameters with default value, right?